lib/resque/worker.rb

```diff
@@ -228,7 +228,7 @@ def work(interval = 5.0, &block)
           break if interval.zero?
           log_with_severity :debug, "Sleeping for #{interval} seconds"
           procline paused? ? "Paused" : "Waiting for #{queues.join(',')}"
-          sleep interval
+          @signal_handler.handle_signal(interval)
```

Comment:
can we do a non-blocking `select` so it doesn't sleep and just returns if there was no signal yet? and keep the `sleep` here where it was